src/main/java/com/pokegoapi/api/pokemon/Pokemon.java

```diff
@@ -55,14 +57,31 @@ public Result transferPokemon() throws LoginFailedException, RemoteServerExcepti
 		return response.getResult();
 	}
 
+	public NicknamePokemonResponse.Result renamePokemon(String nickname) throws LoginFailedException, RemoteServerException {
+		NicknamePokemonMessage reqMsg = NicknamePokemonMessage.newBuilder().setPokemonId(getId()).setNickname(nickname).build();
+
+		ServerRequest serverRequest = new ServerRequest(RequestType.NICKNAME_POKEMON, reqMsg);
+		pgo.getRequestHandler().request(serverRequest);
+		pgo.getRequestHandler().sendServerRequests();
+
+		NicknamePokemonResponse response;
+		try {
+			response = NicknamePokemonResponse.parseFrom(serverRequest.getData());
+		} catch (InvalidProtocolBufferException e) {
+			return null;
```

Comment:
The method throws RemoteServerException.
... `throws LoginFailedException, RemoteServerException {`